app/javascript/psulib_blacklight/javascripts/availability.js

```diff
@@ -383,29 +379,67 @@ function groupByLibrary(holdings) {
-
-        location = `<a data-type="ill-link" ${shared} href="#">${spinner}Request Scan - Penn State Users</a><br>
-                    <a href="https://psu.illiad.oclc.org/illiad/upm/lending/lendinglogon.html">Request Scan - Guest</a><br>
-                    <a data-type="aeon-link" ${shared} data-item-id="${item.itemID}" data-item-location="${aeonLocation}" href="#">${spinner}View in Special Collections</a>`;
-    } else if (['UP-SPECCOL'].includes(item.libraryID)) {
-        var aeonLocation = mapLocation(allLocations, item);
-
-        location = `${aeonLocation}<br><a data-type="aeon-link" data-catkey="${item.catkey}" data-call-number="${item.callNumber}" data-link-type="archival-material" data-item-type="${item.itemTypeID}" data-item-id="${item.itemID}" data-item-location="${aeonLocation}" href="#">${spinner}Request Material</a>`;
-    } else {
-        location = mapLocation(allLocations, item);
+    // Location information presented to the user is different based on a few scenarios
+    // First, if it's related to ILL
+    if (item.locationID in illiadLocations) {
+        illLocation = `<a 
+                            data-type="ill-link" 
+                            data-catkey="${item.catkey}" 
+                            data-call-number="${item.callNumber}" 
+                            href="#"
+                        >${spinner}Copy unavailable, request via Interlibrary Loan</a>`;
+
+        return illLocation;
+    }
+
+    // AEON
+    if (['ARKTHESES', 'AH-X-TRANS'].includes(item.locationID)) {
+        var aeonLocationText = mapLocation(allLocations, item.locationID);
+
+        var aeonLocation = `<a 
+                                data-type="ill-link" 
+                                data-catkey="${item.catkey}" 
+                                data-call-number="${item.callNumber}" 
+                                data-link-type="archival-thesis" 
+                                data-item-type="${item.itemTypeID}" 
+                                href="#">${spinner}Request Scan - Penn State Users</a><br>
+                            <a 
+                                href="https://psu.illiad.oclc.org/illiad/upm/lending/lendinglogon.html">Request Scan - 
+                                Guest</a><br>
+                            <a 
+                                data-type="aeon-link" 
+                                data-catkey="${item.catkey}"
+                                data-call-number="${item.callNumber}"
+                                data-link-type="archival-thesis"
+                                data-item-type="${item.itemTypeID}"
+                                data-item-id="${item.itemID}"
+                                data-item-location="${aeonLocationText}"
+                                href="#">${spinner}View in Special Collections</a>`;
+
+        return aeonLocation;
+    }
+
+    // Special Colletions
+    if (['UP-SPECCOL'].includes(item.libraryID)) {
```

Comment:
@cdmo We need to use the Archival Thesis links for for locations `'ARKTHESES', 'AH-X-TRANS'`, since there might be such location for the `UP-SPECCOL` library, we should either check if location is not one of the archival thesis ones here or we do it in a else if block or change the order of the if blocks?

Reply:
So, https://github.com/psu-libraries/psulib_blacklight/pull/448/files#diff-ae31f8e714d91a5996b94e9240f9256eR398-R423 (lines 398 to 423) are always going to be executed first.

The code in `429-aeon-links-non-thesis` has logic that treats this as an OR as well, i.e, [an else if followed by an else if](https://github.com/psu-libraries/psulib_blacklight/blob/429-aeon-links-non-thesis/app/javascript/psulib_blacklight/javascripts/availability.js#L393-L403). If the first scenario is met on your branch (i.e., ['ARKTHESES', 'AH-X-TRANS'] is present) it won't get to the next scenario regarding ['UP-SPECCOL'] right?

Reply:
> If the first scenario is met on your branch (i.e., ['ARKTHESES', 'AH-X-TRANS'] is present) it won't get to the next scenario regarding ['UP-SPECCOL'] right?

correct ☝️

Reply:
> We need to use the Archival Thesis links for for locations 'ARKTHESES', 'AH-X-TRANS'

That's what [this](https://github.com/psu-libraries/psulib_blacklight/pull/448/files#diff-ae31f8e714d91a5996b94e9240f9256eR399) (line 399) is doing though

Reply:
Yes, it does but for holdings with `UP-SPECCOL` as library and `['ARKTHESES', 'AH-X-TRANS']` as currentLocationID, [this](https://github.com/psu-libraries/psulib_blacklight/pull/448/files#diff-ae31f8e714d91a5996b94e9240f9256eR426) (line 426) should not be true, it should only do line `399`. So sth like this
`if (['UP-SPECCOL'].includes(item.libraryID) && !(['ARKTHESES', 'AH-X-TRANS'].includes(item.locationID))) {` should address those cases, or another way to filter out those cases.

Reply:
per our convo with @cdmo this is correct, it should work correctly since once it finds an archival thesis, it returns. I did not see that return originally.